### PR TITLE
consolidate etcd version per file

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/kind cleanup
/priority backlog
/sig release
/sig testing
/sig cluster-lifecycle

**What this PR does / why we need it**:  consolidate etcd version per file

**Which issue(s) this PR fixes**: helps #79366

**Special notes for your reviewer**:

/assign @neolit123 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -403,9 +403,9 @@ var (
 	SupportedEtcdVersion = map[uint8]string{
 		12: "3.2.24",
 		13: "3.2.24",
-		14: "3.3.10",
-		15: "3.3.10",
-		16: "3.3.10",
+		14: DefaultEtcdVersion,
+		15: DefaultEtcdVersion,
+		16: DefaultEtcdVersion,
```
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants.go]: i'd leave this as is, even if it duplicates the existing default version.
/cc @rosti

File: test/e2e/framework/nodes_util.go [hunk 2]
```diff
@@ -62,7 +64,7 @@ func etcdUpgradeGCE(targetStorage, targetVersion string) error {
 		os.Environ(),
 		"TEST_ETCD_VERSION="+targetVersion,
 		"STORAGE_BACKEND="+targetStorage,
-		"TEST_ETCD_IMAGE=3.3.10-1")
+		"TEST_ETCD_IMAGE="+etcdImage)
```
Comment (neolit123) [inline test/e2e/framework/nodes_util.go]: it feels like the framework changes should have been a separate PR.

File: cmd/kubeadm/app/constants/constants.go (hunk 1 above)
Comment (yastij) [reply cmd/kubeadm/app/constants/constants.go]: I'm reverting this, if we bump `DefaultEtcdVersion` this would bump the etcd version for 14,15,16

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve
/hold
feel free to hold cancel after a day if no objections come trough.

**Pull request merged**